### ISSUE TITLE
[Feat] cloze support within mermaid diagram

### ISSUE THREAD

**Issue by mkschulze** [st3v3nmw/obsidian-spaced-repetition#80]
I'm not sure if this would even be possible. But it would be great to have close tags within a mermaid diagram.

like
```markdown
This is a question and the answer is a mermaid chart ideally with cloze support.
?
```mermaid 
graph TD
    ==Start== --> ==Stop==
```
![Bildschirmfoto 2021-05-16 um 16 12 20](https://user-images.githubusercontent.com/3390820/118400337-7ff86400-b661-11eb-8e14-24fd3a30be36.png)

**Comment by st3v3nmw** [comment]
This looks like the image occlusion feature request #34 only on mermaid charts.

> ```gfm
>  ==Start== --> ==Stop==
> ```

After a cursory look on the mermaid API, I don't think this would be achieved this way. Maybe have a class & some styling to do the occlusions (https://mermaid-js.github.io/mermaid/#/flowchart?id=styling-and-classes, https://mermaid-js.github.io/mermaid/#/theming).
Although this would be _very_ hacky.

**Comment by mkschulze** [comment]
So you mean defining a class and putting the chart with hidden fields in the question and the chart without that class into the answer, right? That should work I guess yes.

**Comment by st3v3nmw** [comment]
Yeah, but a little more automated.

You'd define one mermaid chart as usual, then do something like:
```markdown
class nodeId1,nodeId2,...,nodeIdN hidden;
```

Then the plugin would prompt using those in turn like the usual image occlusions/cloze deletions.

**Comment by mkschulze** [comment]
Ok, so we wouldn't need to change anything in here and the issue could be closed?

**Comment by st3v3nmw** [comment]
Don't close it yet since the plugin doesn't yet have this part of the functionality:

> Then the plugin would prompt using those in turn like the usual image occlusions/cloze deletions.

**Comment by mkschulze** [comment]
would still be nice I think :)

**Comment by LeonelRFF** [comment]
I don't think it would help much, but something that occurred to me is that every time a diagram is created, two more of the same could be generated, but when viewing them, the clue or answer would be shown.
Within the diagram that is created, it is added with quotation marks (to take it only as text)

What the user would create would be something like this:

```
graph TD
    A{"{{a;;question cloze 00;;hint}}"} --> B[question cloze 01→ hint]
    A --> C[question cloze 02→ hint]
```
and in the background, two variants would be created, which are the ones that would be seen when reviewing.

#### Cloze hidden

```mermaid
graph TD
    A{question cloze 00→ hint} --> B[question cloze 01→ hint]
    A --> C[question cloze 02→ hint]
```


#### hint showed

```mermaid
graph TD
    A{answer cloze 00→ hint showed} --> B[answer cloze 01→ hint showed]
    A --> C[answer cloze 02→ hint showed]
```


#### Cloze and hint showed

```mermaid
graph TD
    A{answer cloze 00 showed→ hint showed} --> B[answer cloze 01 showed→ hint showed]
    A --> C[answer cloze 02 showed→ hint showed]
```
